hoomd/VectorVariant.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2009-2023 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#include "VectorVariant.h"
+#include <pybind11/stl.h>
+
+namespace hoomd
+    {
+//* Trampoline for classes inherited in python
+class VectorVariantBoxPy : public VectorVariantBox
+    {
+    public:
+    // Inherit the constructors
+    using VectorVariantBox::VectorVariantBox;
+
+    // trampoline method
+    array_type operator()(uint64_t timestep) override
+        {
+        PYBIND11_OVERLOAD_NAME(array_type,       // Return type
+                               VectorVariantBox, // Parent class
+                               "__call__",       // name of function in python
+                               operator(),       // Name of function in C++
+                               timestep          // Argument(s)
+        );
+        }
+    };
+
+namespace detail
+    {
+
+// This testVariantCall function allows us to test that Python custom vector
+// variants work properly in C++. This ensures we can test that the function
+// itself can be called in C++ when defined in Python.
+
+/// Method to enable unit testing of C++ variant calls from pytest
+std::array<Scalar, 6> testVectorVariantBoxCall(std::shared_ptr<VectorVariantBox> t, uint64_t step)
+    {
+    return (*t)(step);
+    }
+
+void export_VectorVariantBox(pybind11::module& m)
```

Comment:
```suggestion
void export_VectorVariantBoxClasses(pybind11::module& m)
```
This exports more than `VectorVariantBox`.